### PR TITLE
Update integration tests to use TLS enabled DB/storage

### PR DESCRIPTION
**Pull request by DharmitD**
## The issue resolved by this Pull Request:
Resolves https://issues.redhat.com/browse/RHOAIENG-4254

## Description of your changes:

- Generated certs for Minio and MariaDB external. Stored these certs in secrets and configmaps, and added them to respective deployments as volumes and volume mounts, so that both Minio and MariaDB are TLS enabled.
- Stored the signing certificate in a configmap and applied that in the namespace where DSPA with external connections is being deployed.

## Testing instructions

Follow steps in the GH workflow and accordingly deploy and set up changes from this PR on your cluster. Make sure all DSPA resources come up correctly, check DSPO pod logs to ensure connection to TLS enabled Minio and MariaDB is established.

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/resources/mariadb/certs-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
...Y1B6eHh2MjE0Q2hYRUtoclpSK1BXdk4xT2FwQgpqaXdLc2RhNjhOTlprSU9xZVBidUhZYVN2S09uWGM1UVhCTXJwSUJCNXFRc05BM2lsQUVRWFI5ZkI3SVJNdXlECk1mZUNRakxKRXh4SkpXaStaWVFPN3ZzbW5qY055cHVGa1pqdHp0UDEzQkttdFZ6S2h5RWdvT01HNjdFcUJzY1EKNVpuM0FvSUJBR1JYRExHdnhBVHp3MXZYSHR6RzJOa0ZkUkpUZG4raWk1UVlxc20wNExBakQwUmI0UVJVbmlKRQp0YkZKcmFSdndsZjQ2YmM0SHp6czZ6OHFUSDIvbk5iWXliMDB4MDRPWm5JZVFMWDVZaDgybWJsZUxwUWFGL0w0CnhibFZHQ3hwM1B1b04rZjNGNTZKS3U4RUo5ZEdBVnVlZFM0ZVlNd2psM2FUODU3V3BUUFRiR3MrY1FvV1dva3gKZmtqWnJHdDFxQ3ZmdW51OUdsZlJ4RzB2blh4N3dJd3QyNzRqNnRJWk5QRDd6aTFncHNvYnVWWG5ob1g3ZEdYMAoxclk1aGVrODdyNGExamJkOWtnSkFRWkN1NEpmdGJvYlhXZkJXZUVNTUNqNUdpd01Rd0o3THk5c3VGRDVuZTdDCmhUSmxiK1NUUURmQndtT1NhWS8zS3BsVGxpeVdhVUVDZ2dFQUpoeHdKeE12dHpQZTliVDU5NWFqVDN1aVpsQ1AKVTJIbTRtQUwrWlFZaG9hcGM5WWhnR2dhbDF0ZlJsNmxBd1pSdlFqSlF1enBoM2FSQlpGbkZaVUxCSjZ5QU9LKwozLy9FYUdrSXphU2tablAvUjFteklrVkExeWRHMG5mNTJONFJjdzEzV054R2phVFA3M0hqV2VydnVhNkt4N29jClR5WG42Tm9KSVpTQW8rT0liaU9lUVc1dlc2L0g5d0VTV1QyYmYwY0tGaGxmZDF0SE5OZHgzejFUVnloZ3FjLzEKNlNMbXpBUzdWeXRMeTBGRTF4azlNek9CMThRZFhWenQ3b051cFZjSzhOMmJZdWQxazh1UzloYTRqZ1BKbnY0SAovdGNiZjNna0h2QTV1VnRLWU9hVWc3WDF4NmhOaFRWN2JvdXRKays4ZndGVjZQS2pPRzF1Si9ic293PT0KLS0tLS1FTkQgUlNBIFBSSVZBVEUgS0VZLS0tLS0K
```
Comment (gmfrasca) [inline .github/resources/mariadb/certs-secret.yaml:11]: assuming this is a mock cert, at a minimum we need to add this file to a .gitleaks.toml or else we will get spammed by security scanners

File: .github/resources/apiserver/root-ca-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: root-ca
+  managedFields:
+    - manager: Mozilla
+      operation: Update
+      apiVersion: v1
```
Comment (HumairAK) [inline .github/resources/apiserver/root-ca-configmap.yaml:8]: can you remove all occurrences of `managedFields` in any of the manifests, this is not required

File: .github/resources/minio/cabundle-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - manager: openshift-pipelines-operator
+      operation: Update
+      apiVersion: v1
+    - manager: service-ca-operator
+      operation: Update
+      apiVersion: v1
+      fieldsType: FieldsV1
+  namespace: test-minio
+  labels:
+    app.kubernetes.io/part-of: tekton-pipelines
```
Comment (HumairAK) [inline .github/resources/minio/cabundle-configmap.yaml:17]: not needed

File: .github/resources/minio/deployment.yaml [hunk 4]
```diff
@@ -44,11 +60,18 @@ spec:
             - name: data
               mountPath: /data
               subPath: minio
+            - name: minio-certs
+              mountPath: /.minio/certs
+            - name: cabundle
+              mountPath: /.minio/certs/CAs
+          terminationMessagePolicy: File
```
Comment (HumairAK) [inline .github/resources/minio/deployment.yaml:67]: > terminationMessagePolicy: File

remove

File: .github/resources/apiserver/root-ca-configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+kind: ConfigMap
```
Comment (HumairAK) [inline .github/resources/apiserver/root-ca-configmap.yaml:1]: naming this folder `apiserver` for this file is a bit confusing, we don't deploy apiserver, the DSPO does. Let's name the folder `tls`. 

Also, let's leave a comment for the maintainer on what this rootCA is being used for.

Comment (HumairAK) [reply .github/resources/apiserver/root-ca-configmap.yaml:1]: can you actually create the following instead: 

```yaml
├── resources
│   ├── external-pre-reqs
│   │   ├── kustomization.yaml
│   │   ├── mariadb-secret.yaml               
│   │   ├── minio-secret.yaml
│   │   └── root-ca.yaml
```

* mariadb-secret.yaml has the same contents as [this](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/.github/resources/mariadb/secret.yaml)
*  minio-secret.yaml has the same contents as [this](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/.github/resources/minio/secret.yaml)


then get rid of [this](https://github.com/DharmitD/data-science-pipelines-operator/blob/external-tls/.github/workflows/kind-integration.yml#L117-L130). And deploy them all together via `kustomize build` 

in `kustomization.yaml` above, link the folders to the [mariadb](https://github.com/DharmitD/data-science-pipelines-operator/tree/external-tls/.github/resources/mariadb) and [minio](https://github.com/DharmitD/data-science-pipelines-operator/tree/external-tls/.github/resources/minio) for reference

File: .github/resources/mariadb/certs-secret.yaml (hunk 1 above)
Comment (DharmitD) [reply .github/resources/mariadb/certs-secret.yaml:7]: Done, added all configmaps and secrets containing certs and key to a `.gitleaks.toml`.

**Comment by HumairAK** [comment]
refactor of secrets can be in a follow up pr 

/lgtm
/approve

**Pull request merged**